examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -1001,7 +1012,8 @@ def main(args):
         cur_class_images = len(list(class_images_dir.iterdir()))
 
         if cur_class_images < args.num_class_images:
-            torch_dtype = torch.float16 if accelerator.device.type == "cuda" else torch.float32
+            has_supported_fp16_accelerator = accelerator.device.type == "cuda" or torch.backends.mps.is_available()
+            torch_dtype = torch.float16 if has_supported_fp16_accelerator else torch.float32
```

Comment:
i was updating the code to work for both cuda and mps, and it already does that for cuda

Reply:
you would like me to update it for cuda *and* mps to ask the user?

Reply:
For CUDA and `torch.float16` there's no silent change of the `torch_dtype`. But for MPS, it's the case right? If so, yes, let's explicitly ask the user.

Reply:
no - this is just for generating more prior loss images.

the original code uses a fp16 dtype for any cuda GPU, and fp32 for CPU

Reply:
it could be updated to use the same dtype as the principle weights but i'd rather keep it the same for this particular set of changes, it's a bit of a pain to change this many files :P